tests/integration/features/features_int_test.go

```diff
@@ -127,6 +135,119 @@ var _ = Describe("preconditions", func() {
 
 })
 
+var _ = Describe("Manifest sources", func() {
+	Context("using various manifest sources", func() {
+
+		var (
+			objectCleaner *envtestutil.Cleaner
+			dsciSpec      *dscv1.DSCInitializationSpec
+			namespace     = "default"
+		)
+
+		BeforeEach(func() {
+			objectCleaner = envtestutil.CreateCleaner(envTestClient, envTest.Config, timeout, interval)
+			dsciSpec = newDSCInitializationSpec(namespace)
+		})
+
+		It("should be able to process an embedded template from the default location", func() {
+			// given
+			ns := createNamespace("service-ns")
+			Expect(envTestClient.Create(context.Background(), ns)).To(Succeed())
+			defer objectCleaner.DeleteAll(ns)
+
+			serviceMeshSpec := &dsciSpec.ServiceMesh
+			serviceMeshSpec.ControlPlane.Namespace = "service-ns"
+
+			createService, err := feature.CreateFeature("create-control-plane").
+				For(dsciSpec).
+				Manifests(path.Join(templatesDir, "serverless", "serving-istio-gateways", "local-gateway-svc.tmpl")).
+				UsingConfig(envTest.Config).
+				Load()
+
+			Expect(err).ToNot(HaveOccurred())
+
+			// when
+			By("verifying service is created after applying feature", func() {
+				Expect(createService.Apply()).To(Succeed())
+				service, err := getService("knative-local-gateway", "service-ns")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(service).ToNot(BeNil())
+			})
```

Comment:
The `By` clause here is a bit confusing. If we structure the tests using "gherkin" BDD blocks, then we miss the verification block:

* `given` - setting the test fixture
* `when` - exercise system under test
* `then` - check if your expectations match the reality

So maybe this instead:

```suggestion
			// when
			Expect(createService.Apply()).To(Succeed())

			// then
			service, err := getService("knative-local-gateway", "service-ns")
			Expect(err).ToNot(HaveOccurred())
			Expect(service).ToNot(BeNil())
```

How about testing the service name instead of just its existence (`Expect(service).ToNot(BeNil())`). This would to some extent ensure that the template was processed.

Reply:
740741b added the name check here!